pp/requests/requests.go

```diff
@@ -115,6 +115,58 @@ func ReqGetWalletOzData(walletAddr string) *protos.ReqGetWalletOz {
 	}
 }
 
+// RequestUploadFile request upload file without locating the file at the "path"
+func RequestUploadFile(fileName, fileHash string, fileSize uint64, reqID, ownerWalletAddress string, isEncrypted bool) *protos.ReqUploadFile {
+	utils.Log("fileName~~~~~~~~~~~~~~~~~~~~~~~~", fileName)
+	encryptionTag := ""
+	if isEncrypted {
+		encryptionTag = utils.GetRandomString(8)
+	}
+
+	utils.Log("fileHash~~~~~~~~~~~~~~~~~~~~~~", fileHash)
+
+	p2pFileString := setting.WalletAddress + setting.P2PAddress + ownerWalletAddress + fileHash + header.ReqUploadFile
```

Comment:
this is not very ideal, as it's hard to track what to sign, better make it a function, so we can adjust it in a single place.